monai/transforms/post/array.py

```diff
@@ -936,3 +938,30 @@ def __call__(self, image: NdarrayOrTensor) -> torch.Tensor:
         grads = convert_to_dst_type(grads.squeeze(0), image_tensor)[0]
 
         return grads
+
+
+class DistanceTransformEDT(Transform):
+    """
+    Applies the Euclidean distance transform on the input.
+
+    Either GPU based with CuPy / cuCIM or CPU based with scipy.ndimage.
+    Choice only depends on cuCIM being available.
+    Note that the calculations can deviate, for details look into the cuCIM about distance_transform_edt().
+    """
+
+    backend = [TransformBackends.NUMPY, TransformBackends.CUPY]
+
+    def __init__(self, sampling: None | float | list[float] = None, force_scipy: bool = False) -> None:
+        super().__init__()
+        self.force_scipy = force_scipy
+        self.sampling = sampling
+
+    def __call__(self, img: NdarrayOrTensor) -> NdarrayOrTensor:
+        """
+        Args:
+            img: shape must 2D or 3D for cupy, otherwise no restrictions
```

Comment:
please make sure it can work with this input shape assumption (channel first and the distances are computed within each channel): https://github.com/Project-MONAI/MONAI/blob/f76957eb22c924302666eb9c60f0ac3e5eb4edc6/monai/transforms/post/array.py#L320-L324

Reply:
Sadly CuPy cannot handle 4D inputs, see the tests about that. I made the docs a bit more clear about that and added some text there.
CuPy only accepts 2D and 3D inputs, whereas scipy can handle any input. The transform however is configurable to force the scipy implementation so I guess people can choose scipy if they have a different shaped volume.
Is that good enough?

Reply:
Ideally these should be applied channel wise, that's the shape format most of the monai transform follow. please see the logic here using a for loop and then stack the outputs: https://github.com/Project-MONAI/MONAI/blob/56ca2249541bf18c7d3b4f0218db18767ee8bf65/monai/transforms/intensity/array.py#L685

Reply:
thanks, looks good to me,  running more tests